File: existant/gestFormationWeb/gestion_formation_web/dao/Dao.cs
using System.Linq;
namespace gestion_formation_web.dao
{
    public class Dao
    {
        protected static gest_formationEntities1 ctxt = new gest_formationEntities1();
        public static void Update()
        {
            ctxt.SaveChanges();
        }

        public static string[] getEnumValues(string nomTable, string nomColonne)
        {
            //string enumDesc = ctxt.Database.SqlQuery<string>("desc {0} {1} ", new object[] { nomTable, nomColonne }).SingleOrDefault<string>();
            string enumDesc = ctxt.Database.SqlQuery<string>("select  COLUMN_TYPE from information_schema.columns where table_name = {0} AND COLUMN_NAME ={1} limit 1", new string[] { nomTable, nomColonne }).SingleOrDefault<string>();
            
            string enumValues = enumDesc.Substring(enumDesc.IndexOf("(") + 1, (enumDesc.LastIndexOf(")")- enumDesc.IndexOf("(")-1));
            enumValues = enumValues.Replace("'", "");
            string[] valeurs = enumValues.Split(',');
            return valeurs;
        }
    }

}

File: Gestionnaire de formation/src/main/java/fr/esic/solutec/dao/Dao.java
package fr.esic.solutec.dao;

import javax.persistence.EntityManager;
import javax.persistence.EntityManagerFactory;
import javax.persistence.Persistence;
import javax.persistence.Query;

public class Dao {

	static EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("gestionnaire-de-formation") ;
	protected static EntityManager entityManager = entityManagerFactory.createEntityManager();
	
	
	public void close() {
		entityManager.close();
		entityManagerFactory.close();
	}
	
	public static String[] getEnumValues(String nomTable, String nomColonne)
    {
        /* Cette Methode permet d'obtenir le type d'une colonne d'une table  */
		Query query = entityManager.createNativeQuery("select  COLUMN_TYPE from information_schema.columns where table_name = '"+ nomTable + "' AND COLUMN_NAME = '"+nomColonne+"'");
        
		String result = (String)query.getSingleResult();
		
		
		
		String enumValues = result.substring(result.indexOf("(") + 1, result.indexOf(")"));
        enumValues = enumValues.replace("'", "");
        String[] valeurs = enumValues.split(",");
        return valeurs;
    }
	
}
